### PR TITLE
get policies function and test case

### PR DESCRIPTION
**Pull request by alprinf**
PR for iam manager related functions
- created this PR & new branch due to git issue of [previous PR](https://github.com/lks21c/baram/pull/34)
- added `get_policies` and `get_redundant_policies` function with test cases

### CODE REVIEW

File: baram/iam_manager.py [hunk 1]
```diff
@@ -61,3 +62,32 @@ def list_group_policies(self, user_group_name, max_items=100):
...
+        result = [i for i in policies['Policies']]
+        marker = policies['Marker']
+        while True:
+            next_policies = self.cli.list_policies(MaxItems=max_result, Marker=marker)
+            result += next_policies['Policies']
+            if 'Marker' in next_policies:
+                marker = next_policies['Marker']
+            else:
+                break
+        return result
```
Comment (lks21c) [inline baram/iam_manager.py]: let me check this. 
1. Does `marker` always return value whether result has next page or not?
2. Does `marker` have only two cases? return empty string or filled string?

If so, I would implement this way.


```
policies = self.cli.list_policies(MaxItems=max_result)
result = policies['Policies']
while len(policies['Marker']) > 0:
    next_policies = self.cli.list_policies(MaxItems=max_result, Marker=policies['Marker'])
    result += next_policies['Policies']
    policies = next_policies
```

File: baram/iam_manager.py [hunk 2]
```diff
@@ -61,3 +62,32 @@ def list_group_policies(self, user_group_name, max_items=100):
...
+
+    def get_redundant_policies(self):
+        """
+        Lists redundant policies that are not attached to any IAM user, group, or role
+        :return:
+        """
+        redundant_policies = [i for i in self.get_policies() if i['AttachmentCount'] == 0]
+        return redundant_policies
```
Comment (lks21c) [inline baram/iam_manager.py]: how about merge two liens?

File: tests/test_iam_manager.py [hunk 3]
```diff
@@ -39,3 +39,12 @@ def test_role_arn(im):
 
     # Then
     assert arn is not None
+
+def test_get_policies(im):
+    pprint(im.get_policies())
```
Comment (lks21c) [inline tests/test_iam_manager.py]: I recommend json dumps with pretty indentation.

**Review by lks21c — CHANGES_REQUESTED** [review]

File: baram/iam_manager.py (hunk 1 above)
Comment (alprinf) [reply baram/iam_manager.py]: 1. If there is no next page, `Marker` parameter is not returned in the response.
2. So there are two cases: 1) there is `Marker` param with string value, 2) no `Marker` param in returned response.

Thank you for your suggestion. In similar context, I modified this part like below thanks to @Baescott 's help.
```
policies = self.cli.list_policies(Scope=scope, MaxItems=max_result)
        result = policies['Policies']
        try:
            while 'Marker' in policies:
                policies = self.cli.list_policies(Scope=scope, MaxItems=max_result, Marker=policies['Marker'])
                result += policies['Policies']
            return result
        except:
            self.logger.info('error')
```

File: baram/iam_manager.py (hunk 2 above)
Comment (alprinf) [reply baram/iam_manager.py]: sure I'll merge them into one line

File: baram/iam_manager.py [hunk 4]
```diff
@@ -61,3 +62,32 @@ def list_group_policies(self, user_group_name, max_items=100):
...
         '''
         return self.cli.list_group_policies(GroupName=user_group_name, MaxItems=max_items)
+
+    def get_policies(self, max_result=1000):
+        """
+        Lists all policies in IAM
+        :return:
+        """
+        policies = self.cli.list_policies(MaxItems=max_result)
+        result = [i for i in policies['Policies']]
```
Comment (Baescott) [inline baram/iam_manager.py]: I think square brackets at here are useless.

File: tests/test_iam_manager.py [hunk 5]
```diff
@@ -39,3 +39,12 @@ def test_role_arn(im):
 
     # Then
     assert arn is not None
+
+def test_get_policies(im):
```
Comment (Baescott) [inline tests/test_iam_manager.py]: Let's apply `Given When Then` format to those.

**Review by Baescott — CHANGES_REQUESTED** [review]

File: baram/iam_manager.py (hunk 1 above)
Comment (lks21c) [reply baram/iam_manager.py]: I think `try-except` is not bad approach to defense your code, but I recommend to apply all method with try-except or not.

File: tests/test_iam_manager.py (hunk 3 above)
Comment (alprinf) [reply tests/test_iam_manager.py]: yes I'll apply json dumps

File: baram/iam_manager.py (hunk 4 above)
Comment (alprinf) [reply baram/iam_manager.py]: agreed. will remove extra brackets

File: tests/test_iam_manager.py (hunk 5 above)
Comment (alprinf) [reply tests/test_iam_manager.py]: sure I will take that format

**Review by lks21c — APPROVED** [review]

**Comment by lks21c** [comment]
@Baescott 

As I've f/u this issue, I merge this PR since your feedback has been reflected.

**Pull request merged**